### PR TITLE
Feature flags & controls for 2FA requirements

### LINKED ISSUE #5792: Add ability to require 2FA for a given project
**What's the problem this feature will solve?**
Currently there is no way to require a certain level of security for a given project.

**Describe the solution you'd like**
Allow a project to require that 2FA is enabled before a new owners/maintainers can be added. This should also prevent users that are not in compliance from doing the activities permitted by their role.

**Additional context**
~Blocked on #5790, otherwise this would allow any attacker to determine the 2FA status for a given user.~

### PR DESCRIPTION
**Pull request by di**
This PR adds a feature flag for the 2FA requirement feature as well as controls for turning on/off the 2FA requirement as a project maintainer.

It also adds feature flags for 2FA mandate availability (i.e., indicated but not enforced) and enforcement.

The various combinations of flags can be tested in dev by modifying `dev/environment` and running `docker-compose up -d` to recreate the web service.

Closes #5792.

### CODE REVIEW

File: dev/environment [hunk 1]
```diff
@@ -43,3 +43,6 @@ VAULT_URL="http://vault:8200"
 VAULT_TOKEN="an insecure vault access token"
 
 GITHUB_TOKEN_SCANNING_META_API_URL="http://notgithub:8000/meta/public_keys/token_scanning"
+TWOFACTORREQUIREMENT_ENABLED=true
```
Comment (ewdurbin) [inline dev/environment:46]: why not admin flags?

**Review by ewdurbin — COMMENTED** [review]
Appears to work as expected through the assumed progression of:

```
TWOFACTORREQUIREMENT_ENABLED=false
TWOFACTORMANDATE_AVAILABLE=false
TWOFACTORMANDATE_ENABLED=false
```

to 

```
TWOFACTORREQUIREMENT_ENABLED=true
TWOFACTORMANDATE_AVAILABLE=true
TWOFACTORMANDATE_ENABLED=false
```

to

```
TWOFACTORREQUIREMENT_ENABLED=true
TWOFACTORMANDATE_AVAILABLE=true
TWOFACTORMANDATE_ENABLED=true
```

But in working with it locally I noticed some things I brought up as comments. Mostly from a user experience/confusion perspective.

File: warehouse/templates/manage/projects.html [hunk 2]
```diff
@@ -65,8 +65,33 @@ <h1 class="page-title">{% trans project_count=projects|length %}Your projects ({
       <div class="split-layout split-layout--no-wrap split-layout--wrap-on-tablet">
         <div>
           <h3 class="package-snippet__title">{{ project.name }}
+            {% if (
```
[the remark this answers is not in the document]
Comment (di) [reply warehouse/templates/manage/projects.html:68]: Great points. I updated this to:
    - Use a more neutral color
    - Remove checkmark
    - Separate badges for 2FA requirement, mandate, and critical project
    - Only show 2FA mandate badge once mandate is enabled
    - Change links to spans for badges that can't be modified

Looks like this now:
image

File: warehouse/manage/views.py [hunk 3]
```diff
@@ -975,21 +976,65 @@ def _key(project):
...
+            "MAX_PROJECT_SIZE": MAX_PROJECT_SIZE,
+            "toggle_2fa_form": self.toggle_2fa_requirement_form_class(),
+        }
+
+    @view_config(
+        request_method="POST",
+        request_param=Toggle2FARequirementForm.__params__,
+        require_reauth=True,
+    )
+    def toggle_2fa_requirement(self):
```
[the remark this answers is not in the document]
Comment (di) [reply warehouse/manage/views.py:1009]: Added `ProjectEvents` when this is enabled/disabled.

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**